soc/arm/renesas_rcar/gen3/pinctrl_soc.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2021 IoT.bzh
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#ifndef ZEPHYR_SOC_ARM_RENESAS_RCAR_GEN3_PINCTRL_SOC_H_
+#define ZEPHYR_SOC_ARM_RENESAS_RCAR_GEN3_PINCTRL_SOC_H_
+#include <devicetree.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <sys/util.h>
+
+struct rcar_pin_func {
+	uint8_t bank:5;      /* bank number 0 - 18 */
+	uint8_t shift:5;     /* bit shift 0 - 28 */
+	uint8_t func:4;      /* choice from 0x0 to 0xF */
+};
+
+#define RCAR_PIN_FLAGS_PULL_SET       BIT(0)
+#define RCAR_PIN_FLAGS_PUEN           BIT(1)
+#define RCAR_PIN_FLAGS_PUD            BIT(2)
+#define RCAR_PIN_FLAGS_DRIVE_SET      BIT(3)
+#define RCAR_PIN_FLAGS_FUNC_SET       BIT(4)
+
+#define RCAR_PIN_PULL_UP      (RCAR_PIN_FLAGS_PULL_SET | RCAR_PIN_FLAGS_PUEN | RCAR_PIN_FLAGS_PUD)
+#define RCAR_PIN_PULL_DOWN    (RCAR_PIN_FLAGS_PULL_SET | RCAR_PIN_FLAGS_PUEN | RCAR_PIN_FLAGS_PUD)
+#define RCAR_PIN_PULL_DISABLE  RCAR_PIN_FLAGS_PULL_SET
+
+/** Type for R-Car pin. */
+typedef struct pinctrl_soc_pin {
+	uint16_t pin;
+	struct rcar_pin_func func;
+	uint8_t flags;
+	uint8_t drive_strength;
+} pinctrl_soc_pin_t;
+
+#define RCAR_IPSR(node_id) \
+	DT_PROP_BY_IDX(node_id, pins, 1)
```

Comment:
`#define RCAR_HAS_IPSR(node_id) DT_PROP_HAS_IDX(node_id, pins, 1)` can likely be added (for improved code readability)